packages/indexer-common/src/indexer-management/actions.ts

```diff
@@ -206,7 +206,44 @@ export class ActionManager {
     })
   }
 
-  private async updateActionStatuses(
+  /**
+   * Mark actions with the given status.
+   * @param actions
+   * @param transaction
+   * @param status
+   * @returns updated actions
+   * @throws error if the update fails
+   */
+  private async markActions(
+    actions: Action[],
+    transaction: Transaction,
+    status: ActionStatus,
+  ): Promise<Action[]> {
+    const updatedActions: Action[] = []
+    for (const result of actions) {
```

Comment:
What do you think about doing a bulk update SQL tx with array of ids instead of for loop that updates one at a time?